core/modules/replica/DatabaseMySQL.cc

```diff
@@ -81,229 +49,67 @@ namespace replica {
-
-bool Row::get(std::string const& columnName, int64_t& value) const { return ::getAsNumber(*this, columnName, value); }
-bool Row::get(std::string const& columnName, int32_t& value) const { return ::getAsNumber(*this, columnName, value); }
-bool Row::get(std::string const& columnName, int16_t& value) const { return ::getAsNumber(*this, columnName, value); }
-bool Row::get(std::string const& columnName, int8_t&  value) const { return ::getAsNumber(*this, columnName, value); }
-
-bool Row::get(size_t columnIdx, float&  value) const { return ::getAsNumber(*this, columnIdx, value); }
-bool Row::get(size_t columnIdx, double& value) const { return ::getAsNumber(*this, columnIdx, value); }
-
-bool Row::get(std::string const& columnName, float&  value) const { return ::getAsNumber(*this, columnName, value); }
-bool Row::get(std::string const& columnName, double& value) const { return ::getAsNumber(*this, columnName, value); }
-
-bool Row::get(size_t columnIdx, bool& value) const {
-    uint8_t number;
-    if (::getAsNumber(*this, columnIdx, number)) {
-        value = number != '0';
-        return true;
-    }
-    return false;
-}
-
-bool Row::get(std::string const& columnName, bool&  value) const {
-    uint8_t number;
-    if (::getAsNumber(*this, columnName, number)) {
-        value = number != '0';
-        return true;
-    }
-    return false;
-}
-
-Row::Cell const& Row::getDataCell(size_t columnIdx) const {
-
-    static std::string const context = "Row::getDataCell()  ";
-
-    if (not isValid()) {
-        throw std::logic_error(context + "the object is not valid");
-    }
-    if (columnIdx >= _index2cell.size()) {
-        throw std::invalid_argument(
-                context + "the column index '" + std::to_string(columnIdx) +
-                "'is not in the result set");
-    }
-    return _index2cell.at(columnIdx);
-}
-
-Row::Cell const& Row::getDataCell(std::string const& columnName) const {
-
-    static std::string const context = "Row::getDataCell()  ";
-
-    if (not isValid()) {
-        throw std::logic_error(context + "the object is not valid");
-    }
-    if (not _name2indexPtr->count(columnName)) {
-        throw std::invalid_argument(
-                context + "the column '" + columnName + "'is not in the result set");
-    }
-    return _index2cell.at(_name2indexPtr->at(columnName));
-}
-
-//////////////////////////////////////////////////
-//                DoNotProcess                  //
-//////////////////////////////////////////////////
-
-DoNotProcess::DoNotProcess(std::string const& name_)
-    :   name(name_) {
-}
-
-/////////////////////////////////////////////
-//                Keyword                  //
-/////////////////////////////////////////////
-
-Keyword const Keyword::SQL_NULL {"NULL"};
-
-Keyword::Keyword(std::string const& name_)
-    :   DoNotProcess(name_) {
+unsigned long Connection::max_allowed_packet() {
+    return 4*1024*1024;
```

Comment:
a comment explaining the logic behind this may be useful in the future.

Reply:
The default value of the parameter depends on a version of the MySQL/MariaDB. The number set here (4 MB) is the most conservative number which existed prior to MuSQL 8.0.2. And it was increased to 64 MB in the later versions. See: https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_max_allowed_packet
A reason why I have this parameters defined here is explained in the header file.
Though, to be honest I myself don't quite like defying it statically here. I would rather force a desired default as a session variable to a server, or pull the current default from the server instead. The former option is not possible with the current version of MariaDB even though the documentation says it's a "SESSION" and "dynamic" variable. Apparently there is a bug in MySQL, or it requires a newer (at least 8.0) version of the database software. The later option (of pulling this default from the database) is also possible. Though, it requires to extend the database implementation (which I have in my code in class DatabaseMySQL) to do so as the very first thing after establishing a connection. Initially I was reluctant to do this. However, after typing all these words above :-) I've managed to convince myself to go for it.

Reply:
I think you're saying you have or will change this to pull the value from the database, which is fine. But, I think having the explanation in the header file is not close enough when using magic numbers - you should at least have a comment directly next to the magic number, at least directing the reader to a comment in the header file, assuming that comment explains the logic behind the number. ...IMHO, YMMV, etc, etc.

Reply:
Agreed. And, actually, this has already been fixed.